src/aap_eda/api/serializers/auth.py

```diff
@@ -14,9 +14,30 @@
 
 from rest_framework import serializers
 
+from aap_eda.services.auth import jwt_access_token, validate_jwt_token
+from aap_eda.services.exceptions import InvalidTokenError
+
 
 class LoginSerializer(serializers.Serializer):
     username = serializers.CharField(required=True)
     password = serializers.CharField(
         required=True, style={"input_type": "password"}
     )
+
+
+class RefreshTokenSerializer(serializers.Serializer):
+    refresh = serializers.CharField(required=True)
+
+    def validate(self, data):
+        try:
+            self.user = validate_jwt_token(data["refresh"], "refresh")
+        except InvalidTokenError as e:
+            raise serializers.ValidationError("Invalid token") from e
+        return data
+
+    def create_token(self):
```

Comment:
@bzwei Is this the correct place for this method, in the serializer? It seems like we need to get the user_id from the jwt_token and use that to generate a new token in the view.